lib/stores/S3Store.js

```diff
@@ -512,10 +527,25 @@ class S3Store extends DataStore {
                             };
                         })
                         .catch((err) => {
-                            throw err;
+                            if (with_parts === false) {
```

Comment:
This if statement is a bit awkward, but `getOffset` is also called from within the s3 store with `with_parts: true` and I wanted to avoid any side effects.